username database connection

paulojblack opened the pull request:
The SQL scripts now add a users table with two test users.

/createuser allows you to add new users to DB and security can be turned on (but is off right now)

/logger now hits the database to query for the user's input instead of reading it from JSON or whatever I was doing before

Cleaned up some stuff here and there.

stanzheng commented:
Reviewed some of this yesterday, getting 500s after creating the user relation model and also populating it with some entries. might be doing something wrong but need to debug before merging this in. 

@paulojblack are there some specific things besides
1. Updating the schema 
2. adding a user
3. navigating to /login

that im missing?

stanzheng commented:
@wbprice or @ttavenner if you could check out this PR sometime in the future and give a LGTM. Probably issues I'm having are with my local user tables not configured correctly and giving me 500.

paulojblack commented:
As far as I can tell it's a little messy but it works, I'll try and be explicit about what I added below and then we can try and sort it out later.

What was added:

1) Username schema with one sample admin entry and one user entry. See database file (sampledata) or look in psql to find the info.

2) /login page - Logging in with the manually uploaded entries works. Note that the login info requests email and password, not username and password.

3) /logout - obvious

4) /createuser - As far as I can tell this properly inserts a new entry to the db. Make SURE that in the scope box you write "admin". Nothing else is formatted though so write whatever.

5) And this is probs extraneous and can be removed but I wanted to play with the front end so I added a button at the bottom right of the landing page that doesn't link anywhere rn. Idk how best to link to stuff inside the site so I left it vacant, I didn't want to make it an href since that seems clumsy compared to however it's been set up with react.

So yeah I'm not sure what is causing the error. I got hung up for a moment because I kept trying to enter the USERNAME rather than the EMAIL of the entries I had created while trying to login. So I can't say if the problem is from a confusing setup or if there really is a bug somewhere.
